text/0202-events-and-logs-api.md

```diff
@@ -0,0 +1,172 @@
+
+It is unspecified whether or under which conditions the same or different Logger instances are returned from this function.
+
+Implementations MUST NOT require users to repeatedly obtain an Logger again with the same name+version+schema_url+event_domain+include_trace_context to pick up configuration changes. This can be achieved either by allowing to work with an outdated configuration or by ensuring that new configuration applies also to previously returned Loggers.
+
+Note: This could, for example, be implemented by storing any mutable configuration in the LoggerProvider and having Logger implementation objects have a reference to the LoggerProvider from which they were obtained. If configuration must be stored per-Logger (such as disabling a certain Logger), the Logger could, for example, do a look-up with its name+version+schema_url+event_domain+include_trace_context in a map in the LoggerProvider, or the LoggerProvider could maintain a registry of all returned Loggers and actively update their configuration if it changes.
+
+The effect of associating a Schema URL with a Logger MUST be that the telemetry emitted using the Logger will be associated with the Schema URL, provided that the emitted data format is capable of representing such association.
+
+### Logger
+
+The Logger is responsible for creating Events and Logs.
+
+Note that Loggers should not be responsible for configuration. This should be the responsibility of the LoggerProvider instead.
+
+#### Logger operations
+
+The Logger MUST provide functions to:
+
+- Create an `Event` and emit it to the processing pipeline.
+  - The API MUST accept an event name as a parameter. The event name provided should be inserted as an attribute with key `event.name`. Care MUST be taken to not override or delete this attribute while the `Event` is created.  This function MAY be named `logEvent`.
+- Create a `Log Record` and emit it to the processing pipeline.
+  - This function MAY be named `logRecord`.
+  - The intended users of this API is Log Appenders.
+
+### Usage
+
+```java
+OpenTelemetry openTelemetry = OpenTelemetry.noop();
+Logger logger = openTelemetry.getLogger("my-scope");
+
+// Using the convenience method to log an event directly
+logger.logEvent("network-changed", 
+                 Attributes.builder().put("type", "wifi").build());
+
+// Using the event builder to log an event
+logger.eventBuilder("page-navigated").build().setAttribute("url", "http://foo.com/bar#new").emit();
+
+// Using the logRecord builder to log a record
+Logger logger = openTelemetry.getLogger("another-scope");
+logger.logRecordBuilder().build().setBody("I am a log message").emit();
+
+```
+
+### Usage in Client-side telemetry
+
+```java
+public void addBrowserEvent(String name, Attributes attributes) {
+   Logger logger = openTelemetry.getLogger("my-scope", "1.0", "browser");
```

Comment:
Since we have multiple string parameters I think we should come up with a different to specify them otherwise it is not clear which one is which. It doesn't affect the OTEPs proposal but we need a better way in implementations.

Reply:
cc: @jack-berg for ideas - maybe the parameters all can go into a separate class for logger configuration.

Reply:
The java implementation will use a builder pattern for this, in line with [traces](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/TracerBuilder.java) and [metrics](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/metrics/MeterBuilder.java), which each argument getting its own named setter.

Reply:
I've now modified the OTEP following the OTEP template and also removed the spec aspects now. 

@jack-berg I followed the example from https://opentelemetry.io/docs/instrumentation/java/manual/#tracing which is not using a builder pattern.

```
Tracer tracer =
    openTelemetry.getTracer("instrumentation-library-name", "1.0.0");
```